### PR TITLE
Use host id in claims instead of node id

### PR DESCRIPTION
**Pull request by alphadose**
Linked to https://github.com/gravitl/netclient/pull/145

### CODE REVIEW

**Comment by theguy951357** [comment]
waiting on changes from https://github.com/gravitl/netclient/pull/145

File: controllers/hosts.go [hunk 1]
```diff
@@ -279,3 +281,96 @@ func deleteHostFromNetwork(w http.ResponseWriter, r *http.Request) {
...
+	} else if authRequest.Password == "" {
+		errorResponse.Message = "W1R3: Password can't be empty"
+		logger.Log(0, request.Header.Get("user"), errorResponse.Message)
+		logic.ReturnErrorResponse(response, request, errorResponse)
+		return
+	}
+	host, err := logic.GetHost(authRequest.ID)
+	if err != nil {
+		errorResponse.Code = http.StatusBadRequest
+		errorResponse.Message = err.Error()
```
Comment (0xdcarns) [inline controllers/hosts.go:329]: all auth request error responses should be the same message ("unauthorized") or something like that from this err down

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: controllers/hosts.go (hunk 1 above)
Comment (alphadose) [reply controllers/hosts.go:329]: done

**Review by theguy951357 — APPROVED** [review]
built server and client on the netmaker and netclient branches

joined network " private"

netclient list shows the right information
pass

File: controllers/hosts.go [hunk 2]
```diff
@@ -26,6 +27,7 @@ func hostHandlers(r *mux.Router) {
 	r.HandleFunc("/api/hosts/{hostid}/networks/{network}", logic.SecurityCheck(true, http.HandlerFunc(deleteHostFromNetwork))).Methods(http.MethodDelete)
 	r.HandleFunc("/api/hosts/{hostid}/relay", logic.SecurityCheck(false, http.HandlerFunc(createHostRelay))).Methods(http.MethodPost)
 	r.HandleFunc("/api/hosts/{hostid}/relay", logic.SecurityCheck(false, http.HandlerFunc(deleteHostRelay))).Methods(http.MethodDelete)
+	r.HandleFunc("/api/hosts/adm/{network}/authenticate", authenticateHost).Methods(http.MethodPost)
```
Comment (0xdcarns) [inline controllers/hosts.go:30]: remove need for {network} parameter for host auth

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: controllers/hosts.go (hunk 2 above)
Comment (alphadose) [reply controllers/hosts.go]: done

File: controllers/hosts.go [hunk 3]
```diff
@@ -279,3 +281,96 @@ func deleteHostFromNetwork(w http.ResponseWriter, r *http.Request) {
 	logger.Log(2, r.Header.Get("user"), fmt.Sprintf("removed host %s from network %s", currHost.Name, network))
 	w.WriteHeader(http.StatusOK)
 }
+
+// swagger:route POST /api/hosts/adm/{network}/authenticate hosts authenticateHost
```
Comment (abhishek9686) [inline controllers/hosts.go:285]: update api route in swagger annotation

File: controllers/hosts.go [hunk 4]
```diff
@@ -279,3 +281,96 @@ func deleteHostFromNetwork(w http.ResponseWriter, r *http.Request) {
...
+//
+// Authenticate to make further API calls related to a network.
+//
```
Comment (abhishek9686) [inline controllers/hosts.go:288]: change description

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: controllers/hosts.go (hunk 3 above)
Comment (alphadose) [reply controllers/hosts.go]: done

File: controllers/hosts.go (hunk 4 above)
Comment (alphadose) [reply controllers/hosts.go:288]: done

**Comment by theguy951357** [comment]
retesting

**Review by 0xdcarns — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
made nodes and server off of branches
joined network
netclient list shows the right information
pass

**Pull request merged**